lib/ask/runtime/channel_broker.ex

```diff
@@ -348,14 +342,12 @@ defmodule Ask.Runtime.ChannelBroker do
   end
 
   @impl true
-  def handle_info(
-        :timeout,
-        %{channel_id: channel_id, config: %{to_db_operations: to_db_operations}} = state
-      ) do
-    info("handle_info[timeout]", channel_id: channel_id, to_db_operations: to_db_operations)
+  def handle_info(:timeout, %{channel_id: channel_id} = state) do
+    info("handle_info[timeout]", channel_id: channel_id)
+
+    # TODO: delete state from agent if both pqueue & active_contacts are empty!
```

Comment:
This `TODO` seems easily solvable, `pqueue` and `active_contacts` can be checked here and then call `ChannelBrokerAgent.clear()`. Following the code the recovery should still work ok, there's something I'm missing?

Reply:
Yes. I'm wondering if there could be any race condition, but we only ever read/update/save the state from the channel broker process which should never run in parallel: each message should be handled sequentially.

We should however only delete the state for the channel, not clear the agent :)